### PR TITLE
jidea-157 Support ISO codes as country parameter to daedalus

### PR DESCRIPTION
**Pull request by EmmaLRussell**
If the country parameter passed to the daedalus method is a string, a new helper `country_name_from_arg` checks if it matches either iso2 or iso3 style codes (2 or 3 uppercase letters). If so, it looks up and returns the country name which the rest of the daedalus method works with. An error is thrown if the code is not supported.

### CODE REVIEW

File: R/country_helpers.R [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#'
+#' @param country A string which could be a country name, or an ISO2 or ISO3
+#' code.
+#'
+#' @return A supported country name
+#' @keywords internal
+country_name_from_arg <- function(country) {
+  lookup_country_name_from_code <- function(code, code_list) {
+    # check code is known, and lookup country name from idx
+    rlang::arg_match(code, code_list)
```
Comment (richfitz) [inline R/country_helpers.R:14]:
```suggestion
    code <- rlang::arg_match(code, code_list)
```

File: R/country_helpers.R [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    idx <- match(code, code_list)
+    daedalus::country_names[[idx]]
+  }
+
+  # regex check for ISO2
+  if (any(grep("^[A-Z]{2}$", country))) {
+    country <- lookup_country_name_from_code(
+      country, daedalus::country_codes_iso2c
+    )
+  }
```
Comment (richfitz) [inline R/country_helpers.R:24]: Suggest if/else if/else as these three block are exclusive

File: R/daedalus.R [hunk 3]
```diff
@@ -122,7 +122,7 @@ daedalus <- function(country,
   # NOTE: names are case sensitive
   checkmate::assert_multi_class(country, c("daedalus_country", "character"))
   if (is.character(country)) {
-    country <- rlang::arg_match(country, daedalus::country_names)
+    country <- country_name_from_arg(country)
```
Comment (richfitz) [inline R/daedalus.R:125]: Perhaps move this into daedalus_country so that all uses of this benefit from the ISO lookup

Comment (richfitz) [reply R/daedalus.R:125]: Plus add something to the docstring on line 5 (and also in daedalus_country if you change it there) and run `devtools::document()` to regenerate the .Rd files

**Review by richfitz — CHANGES_REQUESTED** [review]

File: R/class_country.R [hunk 4]
```diff
@@ -72,14 +73,14 @@ new_daedalus_country <- function(name, parameters) {
 #' # using assignment operators
 #' x$contact_matrix <- matrix(99, 4, 4)
 #' x
-daedalus_country <- function(name,
+daedalus_country <- function(country,
```
Comment (EmmaLRussell) [inline R/class_country.R:76]: "name" didn't seem right for this parameter any more

File: tests/testthat/test-class_daedalus_country.R [hunk 5]
```diff
@@ -20,13 +20,27 @@ test_that("class <daedalus_country>: basic expectations", {
...
+  checkmate::expect_list(countries, "daedalus_country")
+})
+
+test_that("class <daedalus_country>`: works for all country ISO3 codes", {
+  expect_no_condition({
+    countries <- lapply(daedalus::country_codes_iso3c, daedalus_country) # nolint
+  })
+  checkmate::expect_list(countries, "daedalus_country")
+})
+
```
Comment (EmmaLRussell) [inline tests/testthat/test-class_daedalus_country.R:43]: Maybe this is overkill, but it's following the pattern!

**Review by richfitz — APPROVED** [review]

**Pull request merged**